webui/src/lib/components/repository/ObjectsDiff.jsx

```diff
@@ -0,0 +1,138 @@
+    for (const ext of supportedReadableFormats) {
+        if (path.endsWith("." + ext)) {
+            return true;
+        }
+    }
+    return false;
+}
+
+const ContentDiff = ({repoId, path, leftRef, rightRef, leftSize, rightSize, diffType}) => {
+    const left = leftRef && useAPI(async () => objects.get(repoId, leftRef, path),
+        [repoId, leftRef, path]);
+    const right = rightRef && useAPI(async () => objects.get(repoId, rightRef, path),
+        [repoId, rightRef, path]);
+
+    if ((left && left.loading) || (right && right.loading)) return <Loading/>;
+    const err = (left && left.error) || (right && right.err);
+    if (err) return <Error error={err}/>;
+
+    return <div>
+        <span><DiffSizeReport leftSize={leftSize} rightSize={rightSize} diffType={diffType}/></span>
+        <ReactDiffViewer
+            oldValue={left && left.response}
+            newValue={right && right.response}
+            splitView={false}
+            compareMethod={DiffMethod.LINES}/>
+    </div>;
+}
+
+const StatDiff = ({left, right, diffType}) => {
+    switch (diffType) {
+        case 'changed':
+            if (!left && !right) return <Error error={"Invalid input"}/>;
+            break;
+        case 'added':
+            if (!right) return <Error error={"Invalid input: right is missing"}/>;
+            break;
+        case 'removed':
+            if (!left) return <Error error={"Invalid input: left is missing"}/>;
+            break;
+        default:
+            return <Error error={"Unknown diff type: " + diffType}/>;
+    }
+    const rightSize = right && right.size_bytes;
+    const leftSize = left && left.size_bytes;
+    return <>
+        <div className={"stats-diff-block"}>
+            <DiffSizeReport leftSize={leftSize} rightSize={rightSize} diffType={diffType}/>
+        </div>
+    </>;
+}
+
+const DiffSizeReport = ({leftSize, rightSize, diffType}) => {
+    let diffVerbSpan;
+    let size;
+    switch (diffType) {
+        case 'changed':
+            let sizeDiff = leftSize - rightSize;
+            if (sizeDiff < 0) {
+                size = humanSize(Math.abs(sizeDiff));
+                diffVerbSpan = <span className={"added"}>added </span>
+            } else {
+                size = humanSize(sizeDiff);
+                diffVerbSpan = <span className={"removed"}>removed </span>
+            }
+            break;
+        case 'added':
+            size = humanSize(rightSize)
+            diffVerbSpan = <span className={"added"}>added </span>
+            break;
+        case 'removed':
+            size = humanSize(leftSize)
+            diffVerbSpan =  <span className={"removed"}>removed </span>
+            break;
+        default:
+            return <Error error={"Unknown diff type: " + diffType}/>;
+            break;
+    }
+    return <div>
+                {diffVerbSpan}
+                <span className={"diff-size"}>{size}</span>
+                <span> in size</span>
+            </div>;
```

Comment:
suggestion if it make it clear
```suggestion
    let label = diffType;
    let size;
    switch (diffType) {
        case 'changed':
            const sizeDiff = leftSize - rightSize;
            if (sizeDiff < 0) {
                size = -sizeDiff;
                label = "added";
            } else {
                size = sizeDiff;
                label = "removed";
            }
            break;
        case 'added':
            size = rightSize;
            break;
        case 'removed':
            size = leftSize;
            break;
        default:
            return <Error error={"Unknown diff type: " + diffType}/>;
    }
    return <div>
                <span className={label}>{label} </span>
                <span className={"diff-size"}>{humanSize(size)}</span>
                <span> in size</span>
            </div>;
```